### PR TITLE
DX-729 - OTel improvements & Datadog guide

### PR DESCRIPTION
**Pull request by SonjaChevre**
### For internal users - please add a Jira DX PR ticket to the subject!
---
### Preview Link
- OpenTelemetry overview page: https://deploy-preview-3321--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview/
- Datadog page: https://deploy-preview-3321--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog/

### Description
1. Improved OTel overview page
2. Create a new guide for Datadog integration

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/distributed-tracing-overview.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-title: "Distributed Tracing"
+title: "Distributed tracing"
 date: 2023-08-29T16:58:52+03:00
-tags: ["distributed tracing", "open telemetry", "open tracing"]
+tags: ["distributed tracing", "opentelemetry", "opentracing", "traces"]
 description: Overview page to introduce Distirbuted tracing in Tyk
 ---
 
-Distributed tracing is a method of tracking application requests as they flow from frontend devices to backend services and databases. It is used to monitor and troubleshoot requests end-to-end.
+Distributed tracing is a monitoring and diagnostic technique used in software systems to track and visualize the path of requests as they traverse multiple microservices or components. In the context of an API gateway, distributed tracing helps capture and analyse the journey of API requests across various services, providing valuable insights into performance bottlenecks, latency issues, and the overall health of the system.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/distributed-tracing-overview.md]:
```suggestion
Distributed tracing is a monitoring and diagnostic technique used in software systems to track and visualize the path of requests as they traverse multiple microservices or components. In the context of an API gateway, distributed tracing helps capture and analyse the journey of API requests across various services, providing valuable insights into performance bottlenecks, latency issues and the overall health of the system.
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md [hunk 2]
```diff
@@ -1,25 +1,23 @@
 ---
-title: "OpenTelemetry"
+title: "OpenTelemetry Integration with Tyk Gateway"
 date: 2023-08-29T10:28:52+03:00
 tags: ["otel", "opentelemetry"]
 description: Overview page to introduce OpenTelemetry in Tyk
 ---
 
-Since v5.2 of Tyk Gateway, you can now leverage the power of OpenTelemetry, a robust observability framework for cloud-native software. Enhance your API Gateway's monitoring capabilities with distributed tracing.
+Starting from Tyk Gateway version 5.2, you can leverage the power of OpenTelemetry, an open-source observability framework designed for cloud-native software, to enhance your API monitoring with end-to-end distributed tracing.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md]:
```suggestion
Starting from Tyk Gateway version 5.2, you can leverage the power of OpenTelemetry, an open-source observability framework designed for cloud-native software. This enhances your API monitoring with end-to-end distributed tracing.
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
 
-## Enabling OpenTelemetry in Two Steps
-
-### Prerequisites
+This documentation will guide you through the process of enabling and configuring OpenTelemetry in Tyk Gateway. You'll also learn how to customize trace detail levels to meet your monitoring requirements.
 
-Before proceeding with the following steps, ensure that you have configured your OpenTelemetry backend as well as any necessary integrations. To configure your first integration with the OpenTelemetry collector please refer to the relevant page listed within this section:
+For further guidance on configuring your observability back-end, explore our guides for [Datadog]({{< ref "otel_datadog" >}}), [Dynatrace]({{< ref "otel_dynatrace" >}}), [Jaeger]({{< ref "otel_jaeger" >}}), and [New Relic]({{< ref "otel_new_relic" >}}).
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md]:
```suggestion
For further guidance on configuring your observability back-end, explore our guides for [Datadog]({{< ref "otel_datadog" >}}), [Dynatrace]({{< ref "otel_dynatrace" >}}), [Jaeger]({{< ref "otel_jaeger" >}}) and [New Relic]({{< ref "otel_new_relic" >}}).
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md [hunk 4]
```diff
@@ -29,34 +27,35 @@ First, enable OpenTelemetry in the Tyk Gateway. You can do this by editing the T
...
 {{< img src="/img/distributed-tracing/opentelemetry/detailed-tracing-true.png" alt="Detailed Tracing Enabled" width="800px" >}}
 
-By selecting the appropriate setting, you can customize the level of tracing detail to suit your monitoring needs.
+By choosing the appropriate setting, you can customize the level of tracing detail to suit your monitoring needs.
 
-## Understanding your traces
+## Understanding Your Traces
 
-Gaining a comprehensive understanding of your traces requires diving into both the specific operations being performed and the context in which they are executed. This is where attributes and tags come into play. To fully benefit from OpenTelemetry's capabilities, it's essential to grasp the two main types of attributes: **Span Attributes** and **Resource Attributes**.
+Tyk Gateway exposes a helpful set of *span attributes* and *resource attributes* with the generated spans. These attributes helpful to understand your API requests better. To get a clearer picture, you need to look at what specific actions are happening and the context in which they occur. This is span and resource attributes play a significant role.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md]:
```suggestion
Tyk Gateway exposes a helpful set of *span attributes* and *resource attributes* with the generated spans. These attributes provide useful insights for analysing your API requests. A clear analysis can be obtained by observing the specific actions and associated context within each request/response. This is where span and resource attributes play a significant role.
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md [hunk 5]
```diff
@@ -66,32 +65,22 @@ Span attributes are key-value pairs that serve as metadata for individual spans.
 
 ### Resource Attributes
 
-In OpenTelemetry, resource attributes provide contextual information about the entity that produced the telemetry data. These are associated with the service or application as a whole. Conversely, span attributes are associated with specific operations, such as API requests.
+Resource attributes provide contextual information about the entity that produced the telemetry data. Tyk exposes following resource attributes:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md:68]: In the context of Tyk OpenTelemetry what entity(s) produce the telemetry data? Is it a Gateway instance?

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/distributed-tracing-overview.md [hunk 6]
```diff
@@ -1,10 +1,12 @@
 ---
-title: "Distributed Tracing"
+title: "Distributed tracing"
 date: 2023-08-29T16:58:52+03:00
-tags: ["distributed tracing", "open telemetry", "open tracing"]
+tags: ["distributed tracing", "opentelemetry", "opentracing", "traces"]
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/distributed-tracing-overview.md]:
```suggestion
tags: ["distributed tracing", "OpenTelemetry", "OpenTracing", "traces"]
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
+---
+date: 2023-08-29T13:32:12Z
+title: OpenTelemetry with Datadog
+tags: ["distributed tracing", "OpenTelemetry", "Datadog"]
+description: "This guide explains how to setup Tyk Gateway with OpenTelemetry and Datadog to enhance API Observability"
+---
+
+This guide explains how to configure Tyk API Gateway and the OpenTelemetry Collector to collect distributed traces in Datadog. It follows the [reference documentation from Datadog](https://docs.datadoghq.com/opentelemetry/otel_collector_datadog_exporter/?tab=onahost).
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]:
```suggestion
This guide explains how to configure Tyk API Gateway and the OpenTelemetry Collector to collect distributed traces in Datadog. It follows the reference documentation from [Datadog](https://docs.datadoghq.com/opentelemetry/otel_collector_datadog_exporter/?tab=onahost).
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+- [Docker installed on your machine](https://docs.docker.com/get-docker/)
+- Tyk Gateway v5.2.0 or higher
+- OpenTelemetry Collector Contrib [docker image](https://hub.docker.com/r/otel/opentelemetry-collector-contrib). Make sure to use the Contrib distribution of the OpenTelemetry Collector as it is required for the [Datadog exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/datadogexporter). 
+
+
+### Step 1: Configure the OpenTelemetry Collector
+
+You will need:
+* An [API key from Datadog](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token). An API key looks like this: `6c35dacbf2e16aa8cda85a58d9015c3c`.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]:
```suggestion
- An [API key from Datadog](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token). For example, `6c35dacbf2e16aa8cda85a58d9015c3c`. 
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+- [Docker installed on your machine](https://docs.docker.com/get-docker/)
+- Tyk Gateway v5.2.0 or higher
+- OpenTelemetry Collector Contrib [docker image](https://hub.docker.com/r/otel/opentelemetry-collector-contrib). Make sure to use the Contrib distribution of the OpenTelemetry Collector as it is required for the [Datadog exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/datadogexporter). 
+
+
+### Step 1: Configure the OpenTelemetry Collector
+
+You will need:
+* An [API key from Datadog](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token). An API key looks like this: `6c35dacbf2e16aa8cda85a58d9015c3c`. 
+* Your [Datadog site](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site). Examples: `datadoghq.com`, `us3.datadoghq.com`, `datadoghq.eu`.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]:
```suggestion
- Your [Datadog site](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site). Examples are: `datadoghq.com`, `us3.datadoghq.com` and `datadoghq.eu`. 
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+    traces:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [datadog]
+
+```
+
+### Step 2: Configure a test API
+
+If you don't have any APIs configured yet, create a subdirectory called `apps` in the current directory. Create a new file `apidef-hello-world.json` and copy this simple API definition:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]: Maybe add a sentence explaining that example is keyless and only serves for testing and not production?

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    },
+    "active": true
+}
+```
+
+### Step 3: Create the Docker-Compose file
+
+Save the following YAML configuration to a file named `docker-compose.yml`.
+
+```yaml
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md:88]: Should we remove the comments from the docker-compose example?

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+{{< img src="/img/distributed-tracing/opentelemetry/api-gateway-tyk-service-monitoring-datadog.png" alt="Tyk API Gateway service monitoring in Datadog" >}}
+
+
+### Troubleshooting
+
+If you do not observe any traces appearing in Datadog, consider the following steps for resolution:
+
+- Logging: Examine logs from Tyk API Gateway and from the OpenTelemetry Collector for any issues or warnings that might provide insights.
+- Data Ingestion Delays: Be patient, as there could be some delay in data ingestion. Wait for 10 seconds to see if traces eventually appear, as this is the timeout we have configured in the batch processing of the OpenTelemetry collector in step 1.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]:
```suggestion
- Data Ingestion Delays: Be patient, as there could be some delay in data ingestion. Wait for 10 seconds to see if traces eventually appear, as this is the timeout we have configured in the batch processing of the OpenTelemetry collector within step 1.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Many thanks for submitting the PR Sonja and adding the OTel Datadog config doc. I have had a look through and PR looks good to me. Have suggested some minor grammar changes. After changes made and peer review performed will release.

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md (hunk 5 above)
Comment (SonjaChevre) [reply tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview.md:68]: yes, the Gateway instance

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md (hunk 11 above)
Comment (SonjaChevre) [reply tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md:88]: I think the comments are helpful, would like to keep them there.

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md (hunk 10 above)
Comment (SonjaChevre) [reply tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/otel_datadog.md]:
```suggestion
If you don't have any APIs configured yet, create a subdirectory called `apps` in the current directory. Create a new file `apidef-hello-world.json` and copy this very simple API definition for testing purposes:
```

**Review by carolwkc — APPROVED** [review]
LGTM!

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**